cardano-api/src/Cardano/Api/Certificate.hs

```diff
@@ -183,12 +158,137 @@ data StakePoolRelay =
 
 data StakePoolMetadataReference =
      StakePoolMetadataReference {
-       stakePoolMetadataURL  :: URI.URI,
+       stakePoolMetadataURL  :: Text,
```

Comment:
Isn't it safer to use a `URI` to ensure that we've got a valid URI?

Reply:
Yes, but the on-chain format does not. And given that we want to use this type to read the chain as well as write it, then we cannot impose stronger constraints in the types.

We can do optional helpful validation of course.